compiler/noirc_evaluator/src/ssa/ssa_gen/context.rs

```diff
@@ -346,6 +347,20 @@ impl<'a> FunctionContext<'a> {
                         result
                     }
                     BinaryOpKind::ShiftLeft | BinaryOpKind::ShiftRight => {
+                        let dfg = &self.builder.current_function.dfg;
+                        let max_lhs_bits = dfg.get_value_max_num_bits(lhs);
+
+                        if let Some(rhs_const) = dfg.get_numeric_constant(rhs) {
+                            let bit_shift_size = rhs_const.to_u128() as u32;
+
+                            if max_lhs_bits + bit_shift_size <= bit_size {
```

Comment:
max_lhs_bits + bit_shift_size <= bit_size implies that bit_shift_size <= bit_size, which implies that the overflow check will pass at compile time, so there is no need for this.